libnavigation-ui/src/main/java/com/mapbox/navigation/ui/listeners/RouteListener.java

```diff
@@ -55,21 +55,4 @@
    * @since 0.8.0
    */
   void onFailedReroute(String errorMessage);
-
-  /**
-   * Will trigger when a user has arrived at a given waypoint along a {@link DirectionsRoute}.
-   * <p>
-   * For example, if there are two {@link com.mapbox.api.directions.v5.models.LegStep}s, this callback
-   * will be triggered twice, once for each waypoint.
-   *
-   * @since 0.14.0
-   */
-  void onArrival();
-
-  /**
-   * Will trigger when a user has arrived at the final destination on
-   * the {@link DirectionsRoute}. The final destination is the same as the final
-   * {@link com.mapbox.api.directions.v5.models.DirectionsWaypoint} along the route.
-   */
-  void onFinalDestinationArrival();
```

Comment:
Maybe separate discussion but @langsmith had some concerns with the current naming for `ArrivalObserver`'s methods we might want to use this in there cc @kmadsen

Reply:
I think let's do that in a separate pr

Reply:
Feel free to cut a follow up ticket @langsmith so we don't forget 😅

Reply:
Suggest to use onFinalDestinationArrival over onRouteArrival? onStopArrival may not be best either.

Feel free to suggest, the effort was mostly on functionality. Have been thinking these could be better. But would like to see what others think too!
```
onWaypointDeparture
onFinalDestinationArrival
```

Reply:
yea, I had waypoint in mind as well because we refer to waypoints in a lot of nav sdk documentation rather than stops. Waypoints is the Directions API terminology too https://docs.mapbox.com/api/navigation/#waypoint-object

Reply:
> Feel free to cut a follow up ticket @langsmith so we don't forget 😅

Will do

Reply:
> Feel free to cut a follow up ticket @langsmith so we don't forget 😅

https://github.com/mapbox/mapbox-navigation-android/issues/2933